android-app/app/src/main/java/de/dhbw/ka/se/fibo/ui/adding/AddingFragment.java

```diff
@@ -198,34 +206,35 @@ private String getFieldValue(TextView field) {
     }
 
     private boolean checkForRequiredData() throws IllegalArgumentException {
-
-        if (null == store.getText()
-                | store.getText().toString().trim().isEmpty()) {
-            throw new IllegalArgumentException("Store must be set");
-        } else if (null == amount.getText()
-                | amount.getText().toString().trim().isEmpty()) {
-            throw new IllegalArgumentException("Amount must be set");
-        } else if (null == dateText.getText()
-                | dateText.getText().toString().trim().isEmpty()) {
-            throw new IllegalArgumentException("Date must be set");
-        } else if (null == categoriesDropdown.getText()
-                | categoriesDropdown.getText().toString().trim().isEmpty()) {
-            throw new IllegalArgumentException("Category must be set");
-        } else if (null == address.getText()
-                | address.getText().toString().trim().isEmpty()) {
-            throw new IllegalArgumentException("Address must be set");
+        Map<TextInputLayout, String> fieldsToBeChecked = new HashMap<>();
+        if (CashflowType.EXPENSE == newCashFlowType) {
+            fieldsToBeChecked.put(storeLayout, getString(R.string.error_message_store_field));
+        } else {
+            fieldsToBeChecked.put(storeLayout, getString(R.string.error_message_source_field));
         }
-        //others are currently not stored in our database or not required
+        fieldsToBeChecked.put(amountLayout, getString(R.string.error_message_amount_field));
+        fieldsToBeChecked.put(dateTextLayout, getString(R.string.error_message_date_field));
+        fieldsToBeChecked.put(categoriesDropdownLayout, getString(R.string.error_message_category_field));
+        fieldsToBeChecked.put(addressLayout, getString(R.string.error_message_address_field));
 
-        return true;
+        return ActivityUtils.checkValidInput(fieldsToBeChecked);
     }
 
     private void navigateToHome() {
         navController.popBackStack();
     }
 
     private void setUpDateTextField() {
-        dateText.setEnabled(false);
+        dateText.setShowSoftInputOnFocus(false);
+        dateText.setOnFocusChangeListener((v, hasFocus) -> {
+            if (v.isInTouchMode() && hasFocus) {
+                v.performClick();
+            }
+        });
+
+        dateText.setOnClickListener(v ->
+                datePicker.show(requireActivity().getSupportFragmentManager(), "datePick"));
+
         dateText.setTextColor(requireContext().getColor(R.color.mainTextColor));
         binding.dateLayout.setEndIconOnClickListener(v -> datePicker.show(requireActivity().getSupportFragmentManager(), "datePick"));
     }
```

Comment:
```suggestion
        dateText.setFocusable(View.NOT_FOCUSABLE);
        dateText.setOnClickListener(showDatePicker());
        binding.dateLayout.setEndIconOnClickListener(showDatePicker());
    }

    private View.OnClickListener showDatePicker() {
        return (view) -> datePicker.show(requireActivity().getSupportFragmentManager(), "datePick");
    }
    }
```

Can be shortened to this